src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,190 @@
+            Task curentTask = parser.parseDeadline();
+            taskList.add(curentTask);
+            ui.addPrintStatement("added:\t" + ui.displayTask(curentTask));
+        } catch (ParserException ex) {
+            ui.addPrintStatement(ex.getMessage());
+        }
+
+
+    }
+
+    public void addEvent() {
+        try {
+            Task curentTask = parser.parseEvent();
+            taskList.add(curentTask);
+            ui.addPrintStatement("added:\t" + ui.displayTask(curentTask));
+        } catch (ParserException ex) {
+            ui.addPrintStatement(ex.getMessage());
+        }
+    }
+
+    public void deleteTask() {
+        if (taskList.isEmpty()) {
+            ui.addPrintStatement("The list is empty!");
+            return;
+        }
+        try {
+            // remove the current task
+            Task selectedTask = taskList.get(parser.getIndex());
+            taskList.remove(parser.getIndex());
+            ui.displayAction("Deleting selected task!", selectedTask);
+        } catch (IndexOutOfBoundsException ex) {
+            ui.addPrintStatement("Please enter a valid index!");
+        }
+    }
+
+    public void findTask() {
+        ui.addPrintStatement("finding task!");
+        String name = parser.getTaskName();
+        List<Task> tasks = taskList.findTasks(name);
+
+        ui.addPrintStatement("Found:");
+        for (int i = 0; i < tasks.size(); i++) {
+
+            String index = Integer.toString(i + 1);
+            Task selectedTask = tasks.get(i);
+            ui.addPrintStatement(index + " " + ui.displayTask(selectedTask));
+        }
+    }
+
+    public String start() {
+        ui.displayGreetings();
+        return ui.getPrintStatement();
+    }
+
+    public void run(String input) {
+
+        parser.update(input);
+        // there is no input
+        switch (parser.getCommandString()) {
+            case "bye":
+                ui.displayGoodbye();
+                storage.saveTasks();
+                Platform.exit();
+                System.exit(0);
+                break;
+            case "list":
+                listTasks();
+                break;
+            case "mark": {
+                markTaskAsDone();
+                break;
+            }
+            case "unmark": {
+                unmarkTask();
+                break;
+            }
+            case "todo": {
+                addTodo();
+                break;
+            }
+            case "deadline": {
+                addDeadline();
+                break;
+            }
+            case "event": {
+                addEvent();
+                break;
+            }
+            case "delete": {
+                deleteTask();
+                break;
+            }
+            case "find": {
+                findTask();
+                break;
+            }
+            default:
+                ui.addPrintStatement("Please enter a suitable task!");
+        }
+
+    }
+
+    public String getResponse(String input) {
```

Comment:
can consider adding javadocs / comments to describe what this method does

Reply:
Seconded. I like how you used encapsulation to pass the functionality to the `Ui` class, but it is not immediately obvious what this method does.